### PR TITLE
Closes #150; Concise alternative to FieldsWhere.kt

### LINKED ISSUE #150: Shorten usage of Fields in include, where, and orderBy API functions
### Problem

The suggested usage of `Fields` in the library results in too long statements due to redundancy of using `Fields.`. For example,

```kotlin
.where((Fields.Email.Address startsWith "a") and (Fields.Email.Address endsWith "gmail.com"))
```

While this is the only way `Fields` can be used by Java users, we can shorten it for Kotlin users. Actually, Kotlin users can already remove the redundant calls to `Fields.` this by using `run` on it,

```kotlin
.where(Fields.run { (Email.Address startsWith "a") and (Email.Address endsWith "gmail.com") })
```

However, it is not obvious and requires consumers to still use `Fields.` once.

Consumers could also import all properties of `Fields`. However, it could cause a lot of name clashes between the name of the field (e.g. Email) and the name of the entity (e.g. Email).

### Solution

Remove the need for consumers to use `Fields.run` (and other equivalent field sets), in  `include`, `where`, and `orderBy` API functions.

Usage for Kotlin users would look like,

```kotlin
.where{ (Email.Address startsWith "a") and (Email.Address endsWith "gmail.com") }
```

This also gets rid of the need for consumers to know that they should be using `Fields` to begin with, even though it is mentioned in the documentation.

We can take this a step further by providing extensions for a field sets,

```kotlin
.where{ Email { (Address startsWith "a") and (Address endsWith "gmail.com") } }
```

And even further by providing extensions for individual fields,

```kotlin
.where{ Email.Address { startsWith("a") and endsWith("gmail.com") } }
```

### Checklist

- [x] Add the extra functions to all APIs
- [x] Update sample Kotlin code in API class docs
- [x] Update all relevant howto pages
- [x] Update README
- [x] Update medium and dev.to blog code snippets
- [x] Update sample app usages

### PR DESCRIPTION
**Pull request by JayaSuryaT**
This PR brings in a more concise and maintainable way of doing what `contacts.core.util.FieldsWhere.kt` extensions are doing.

More details are discussed in here #162 and here #150.

> **TLDR**: There were a bunch of handwritten extension methods on concrete `Field` and `FieldSet` types, which enabled more idiomatic and _DSLesk_ API surface. Now expressing all of those methods using Kotlin magic (invoke operator function overloads and lambdas) in a more concise way.

As discussed prior, the new extensions are added in the `contacts.core.Where.kt` file and the `contacts.core.util.FieldsWhere.kt` file has been removed in favor of the new extensions.

### CODE REVIEW

File: core/src/main/java/contacts/core/Where.kt [hunk 1]
```diff
@@ -308,6 +308,19 @@ infix fun <T : Field> Where<T>.or(where: Where<T>?): Where<T> = if (where != nul
...
+ * operators on a single [Field].
+ */
+inline operator fun <F : Field> F.invoke(where: F.() -> Where<F>): Where<F> = where(this)
+
+/**
+ * A shorthand extension function to ease the construction of [Where] instances with (multiple)
+ * operators on multiple fields of a [FieldSet].
+ */
+inline operator fun <T : Field, FS : FieldSet<T>> FS.invoke(where: FS.() -> Where<T>): Where<T> =
+    where(this)
```
Comment (vestrel00) [inline core/src/main/java/contacts/core/Where.kt:322]: 10/10 implementation and documentation. Thank you!

File: core/src/main/java/contacts/core/util/FieldsWhere.kt [hunk 2]
```diff
@@ -1,570 +0,0 @@
-@file:Suppress("FunctionName")
```
Comment (vestrel00) [inline core/src/main/java/contacts/core/util/FieldsWhere.kt:1]: Really good thing that I asked [about this on Reddit](https://www.reddit.com/r/Kotlin/comments/rtxfdm/comment/hr19srb/?utm_source=share&utm_medium=web2x&context=3)! You have literally reduced ~500 lines of code down to 2 lines. I have no words. I learned something from this! I don't know everything about Kotlin magic!

I have given you a gold medal for this. Thank you ❤️ 

Screen Shot 2022-01-03 at 3 32 31 PM

I want to give you 50,000 points but I don't have money 😄

**Review by vestrel00 — APPROVED** [review]

File: core/src/main/java/contacts/core/util/FieldsWhere.kt (hunk 2 above)
Comment (JayaSuryaT) [reply core/src/main/java/contacts/core/util/FieldsWhere.kt:1]: That is true, sometimes seemingly random things lead us to some interesting places and situations, this time it was Reddit.

Thank you so much for your appreciation. You are too kind and wholesome.
I'm already chuffed to bits with your responses, they mean a whole lot more to me than Reddit awards. But still, thank you for taking time out and showing appreciation the way you did, be it on Reddit or on GitHub.

This was my first Open Source contribution (other than my own projects), I'm glad it worked out for both of us the way it did ❤️.
